### PR TITLE
Implement calling ASPECT in a way that reads input from stdin.

### PR DESCRIPTION
**Pull request by bangerth**
I couldn't figure out a way to test this with MPI but there is a test for the
non-parallel case and I verified manually that it works with MPI.

### CODE REVIEW

File: tests/dash-dash.prm [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Test that we indeed read from stdin when the input file is given as "--"
+
+# This .prm file is really a sham. Its only purpose is to set up a test that
+# automatically loads a shared library that only starts aspect (again!) with
+# "--" as argument, and when that has finished running terminates the original
+# copy of aspect. In pther words, the original aspect is only used to load a
```
Comment (gassmoeller) [inline tests/dash-dash.prm]: Typo "other"

**Comment by gassmoeller** [comment]
Apart from the typo it looks good. What I like especially is that I can now use ASPECT_SOURCE_DIR in shared libraries of tests (did I get that right from the commit Wolfgang?). And for parameter studies the stdin input is of course great.
One thing I do not understand quite now is what happens if I just type `aspect --` without piping input. Aspect is waiting for input on stdin in this case, but I can only tell it that I am finished writing my parameters by pressing ctrl-D, which is not intuitive to me. I know this case will probably never be used productively, it just confused me a bit. Maybe we can document this behaviour?

**Comment by bangerth** [comment]
@gassmoeller Good point -- yes, a bit awkward to have to type `^D` :-) I've added this to the manual, and I've also mentioned the idea to use this for parameter studies. I don't think I had realized that that is what one can do with this patch ;-)

File: source/main.cc [hunk 2]
```diff
@@ -386,30 +385,81 @@ int main (int argc, char *argv[])
...
+                         MPI_CHAR,
+                         /*root=*/0, MPI_COMM_WORLD);
+            }
+          else
+            {
+              // on this side, read what processor zero has broadcast about
+              // the size of the input file. then create a buffer to put the
+              // text in, get it from processor 0, and copy it to
+              // input_as_string
+	      int size;
```
Comment (tjhei) [inline source/main.cc]: indentation looks messed up, can you run astyle please?

**Comment by tjhei** [comment]
looks good otherwise.

**Comment by bangerth** [comment]
I fixed the indentation issue.

**Pull request merged**